pkg/api/v1/types.go

```diff
@@ -2154,17 +2155,17 @@ func (spec *NutanixNodeSpec) MarshalJSON() ([]byte, error) {
 // VMwareCloudDirectorNodeSpec VMware Cloud Director node settings
 // swagger:model VMwareCloudDirectorNodeSpec
 type VMwareCloudDirectorNodeSpec struct {
-	CPUs             int    `json:"cpus"`
-	CPUCores         int    `json:"cpuCores"`
-	MemoryMB         int    `json:"memoryMB"`
-	DiskSizeGB       *int64 `json:"diskSizeGB,omitempty"`
-	DiskIOPS         *int64 `json:"diskIOPS,omitempty"`
-	Template         string `json:"template"`
-	Catalog          string `json:"catalog"`
-	StorageProfile   string `json:"storageProfile,omitempty"`
-	IPAllocationMode string `json:"ipAllocationMode,omitempty"`
-	VApp             string `json:"vapp"`
-	Network          string `json:"network"`
+	CPUs             int                  `json:"cpus"`
+	CPUCores         int                  `json:"cpuCores"`
+	MemoryMB         int                  `json:"memoryMB"`
+	DiskSizeGB       *int64               `json:"diskSizeGB,omitempty"`
+	DiskIOPS         *int64               `json:"diskIOPS,omitempty"`
+	Template         string               `json:"template"`
+	Catalog          string               `json:"catalog"`
+	StorageProfile   string               `json:"storageProfile"`
+	IPAllocationMode vcd.IPAllocationMode `json:"ipAllocationMode,omitempty"`
```

Comment:
I am fine with the intent behind the PR, but I just randomly wonder if it's a good idea to expose types defined by a third party library. It could also be a great thing. WDYT?